mods/sbz_chem/crusher.lua

```diff
@@ -90,7 +90,15 @@ listring[current_player;main]listring[context;input]listring[current_player;main
                 inv:add_item("output", selected_item)
             else
                 meta:set_string("infotext", "Output inventory full")
-                return power_needed
+            end
+
+            if inv:contains_item("output", "sbz_chem:empty_fluid_cell") then
```

Comment:
this makes automation tricky if it's in the output

because nothing can push to the output

i'd suggest expanding crusher's input to 2 slots instead

Reply:
does it need to be automated this way? water sources are currently renewable in this, so if there's a rightclicking machine or something it could be used with vials on a source between two others perhaps

Reply:
OH, so a node breaker with a bucket (assumbtion: uses on_use) would work? if so then im sorry i didnt notice that

Reply:
unlikely as it stands since the vials use on_place but could be added somehow

Reply:
on_place works too

callbacks that can be fully automated (i can add more):

`on_use`, `on_dig` - node breaker
`on_place` - deployer
`on_punch` - puncher (by far the simplest)

Reply:
just been testing, the deployer is indeed able to place and pick up water, but since it does both, it will keep alternating unless you take out the vial immediately

Reply:
> just been testing, the deployer is indeed able to place and pick up water, but since it does both, it will keep alternating unless you take out the vial immediately

ah, idk if thats an issue

also maybe do what the mtg buckets did(?) where getting water in a bucket is on_use and getting water outside of a bucket is on_place

Reply:
https://github.com/minetest/minetest_game/tree/master/mods/bucket

Reply:
that way logic won't be needed to automate water

Reply:
i mean techically there is `on_pickup` by the item vacuum if you even implemented it

Reply:
pipeworks doesnt implement it i think